org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InlayHintVisitor.java

```diff
@@ -158,12 +158,13 @@ private void resolveInlayHints(IMethodBinding methodBinding, List<Expression> ar
 
 		// not showing the inlay hints when arguments are incomplete,
 		// this is to avoid hint flickering
-		if (arguments.size() < parameterNames.length) {
+		if (!methodBinding.isVarargs() && arguments.size() != parameterNames.length) {
```

Comment:
Nice. I guess this also fixes the case of the hints showing even when using more arguments than the amount of existing parameters.